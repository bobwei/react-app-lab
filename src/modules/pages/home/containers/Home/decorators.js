import compose from 'recompose/compose';
import lifecycle from 'recompose/lifecycle';

export const withData = () => compose(
  lifecycle({
    componentDidMount() {
      const { fetchData } = this.props;
      fetchData({ size: 25 });
    },
  }),
);
