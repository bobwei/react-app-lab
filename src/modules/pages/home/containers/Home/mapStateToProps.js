import R from 'ramda';
import { get } from 'lodash/fp';

const mapStateToProps = state => ({
  data: R.compose(
    R.map(row => ({
      ...R.pick(['id', 'title'])(row),
      photo: get('apps[0].thumbnails.medium.url')(row),
    })),
    ({ list, detail }) => R.map(R.prop(R.__, detail))(list),
    R.pick(['list', 'detail']),
    R.path(['data']),
  )(state),
});

export default mapStateToProps;
