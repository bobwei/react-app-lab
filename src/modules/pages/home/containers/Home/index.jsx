import React from 'react';
import { Jumbotron, Grid } from 'react-bootstrap';
import { Link } from 'react-router';
import compose from 'recompose/compose';

import Button from 'modules/ui/components/Button';

import { withData } from './decorators';

const Home = () => (
  <Grid>
    <Jumbotron>
      <h2>Hello, React App Boilerplate</h2>
      <p>
        This is a react app boilerplate with batteries included.
      </p>
      <p>
        <Button
          bsStyle="default"
          to={{
            pathname: '/admin',
            state: { modal: true },
          }}
          componentClass={Link}
        >
          open Admin Portal in modal
        </Button>
      </p>
    </Jumbotron>
  </Grid>
);

export default compose(
  withData(),
)(Home);
