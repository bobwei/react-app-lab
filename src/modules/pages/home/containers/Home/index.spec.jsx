/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import configureStore from 'stores';
import Home from './index';

it('can render correctly', () => {
  const store = configureStore({});
  const component = renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
