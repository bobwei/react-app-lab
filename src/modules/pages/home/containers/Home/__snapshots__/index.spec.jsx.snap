exports[`test can render correctly 1`] = `
<div
  className="container">
  <div
    className="jumbotron">
    <h2>
      Hello, React App Boilerplate
    </h2>
    <p>
      This is a react app boilerplate with batteries included.
    </p>
    <p>
      <a
        action="push"
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button">
        open Admin Portal in modal
      </a>
    </p>
  </div>
  <div
    className="row">
    <div
      className="col-md-6 col-md-offset-3">
      <form
        className=""
        onSubmit={[Function]}>
        <div
          style={
            Object {
              "margin": "-30px 0 30px",
            }
          }>
          <div
            className="form-group">
            <div>
              <div>
                <input
                  autoFocus={true}
                  className="form-control"
                  name="q"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Search By Name"
                  type="search"
                  value="" />
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div
    className="row" />
</div>
`;
